### ISSUE TITLE
Compilation fails with cl.exe /Zc:wchar_t-

### ISSUE THREAD

**Issue by dandrestor** [onqtam/doctest#206]
## Description
When using the Visual Studio compiler option `/Zc:wchar_t-`, the compiler will replace the type `wchar_t` with a typedef. Thus the following error is generated during compilation:

```
doctest.h(2587): error C2766: explicit specialization; 'const char *doctest::detail::type_to_string<wchar_t>(void)' has already been defined
doctest.h(2585): note: see previous definition of 'type_to_string'
```

* doctest version: **v2.2.3**
* Operating System: **Windows 10**
* Compiler+version: **Visual Studio 14.0**

**Comment by onqtam** [comment]
Thanks for reporting this!

Seems that I can check against a [preprocessor identifier](https://docs.microsoft.com/en-us/cpp/build/reference/zc-wchar-t-wchar-t-is-native-type?view=vs-2017) (some more [info](https://stackoverflow.com/questions/4521252/qt-msvc-and-zcwchar-t-i-want-to-blow-up-the-world))

This should be fixed in version 2.3 when I release it. For now you can uncomment one of those 2 lines.

**Comment by dandrestor** [comment]
Awesome, thanks.
